src/main/java/games/strategy/triplea/delegate/Matches.java

```diff
@@ -1902,7 +1902,7 @@ private Matches() {}
   /**
    * Check if unit meets requiredUnitsToMove criteria and can move into territory.
    */
-  public static Match<Unit> unitHasRequiredUnitsToMove(final Territory t) {
+  public static Match<Unit> unitHasRequiredUnitsToMove(final Territory t, final GameData data) {
```

Comment:
Just curious because I've seen this pattern used a lot throughout the codebase...  Why can't we use the `GameData` instance associated with the `Territory` (or the `Unit`) instead of passing it as a separate parameter?  Won't `t.getData()` (or `unit.getData()`) refer to the same `GameData` instance that is ultimately being passed in as the second argument?

Reply:
Should be able to. Can't say I've really thought about it much.